node/malus/src/variants/store_malicious_available_data.rs

```diff
@@ -0,0 +1,353 @@
+		Some(msg)
+	}
+}
+
+// Meta information about a candidate.
+#[derive(Debug, Encode, Decode)]
+struct CandidateMeta {
+	state: State,
+	data_available: bool,
+	chunks_stored: BitVec<BitOrderLsb0, u8>,
+}
+
+fn store_malicious_available_data(
+	db: &Arc<dyn KeyValueDB>,
+	config: &AvailabilityConfig,
+	candidate_hash: CandidateHash,
+	replaced_chunk_index: usize,
+	n_validators: usize,
+	available_data: AvailableData,
+) -> Result<(), Error> {
+	let mut tx = DBTransaction::new();
+
+	let mut meta = match load_meta(&db, &config, &candidate_hash)? {
+		Some(m) => {
+			if m.data_available {
+				return Ok(()) // already stored.
+			}
+
+			m
+		},
+		None => {
+			let now = SystemTime::now()
+				.duration_since(UNIX_EPOCH)
+				.expect("Getting time should work. QED");
+
+			// Write a pruning record.
+			let prune_at = now + Duration::from_secs(60 * 60);
+			write_pruning_key(&mut tx, &config, prune_at, &candidate_hash);
+
+			CandidateMeta {
+				state: State::Unavailable(BETimestamp(0)),
+				data_available: false,
+				chunks_stored: BitVec::new(),
+			}
+		},
+	};
+
+	let mut chunks = erasure::obtain_chunks_v1(n_validators, &available_data).unwrap();
+	chunks[replaced_chunk_index].fill(42);
+	let branches = erasure::branches(chunks.as_ref());
+
+	let erasure_chunks = chunks.iter().zip(branches.map(|(proof, _)| proof)).enumerate().map(
+		|(index, (chunk, proof))| ErasureChunk {
+			chunk: chunk.clone(),
+			proof,
+			index: ValidatorIndex(index as u32),
+		},
+	);
+
+	for chunk in erasure_chunks {
+		write_chunk(&mut tx, &config, &candidate_hash, chunk.index, &chunk);
+	}
+
+	meta.data_available = true;
+	meta.chunks_stored = bitvec::bitvec![BitOrderLsb0, u8; 1; n_validators];
+
+	write_meta(&mut tx, &config, &candidate_hash, &meta);
+	write_available_data(&mut tx, &config, &candidate_hash, &available_data);
+
+	db.write(tx)?;
+
+	Ok(())
+}
+
+/// Generates an overseer that disputes instead of approving valid candidates.
+pub(crate) struct StoreMaliciousAvailableDataWrapper;
+
+impl OverseerGen for StoreMaliciousAvailableDataWrapper {
+	fn generate<'a, Spawner, RuntimeClient>(
+		&self,
+		connector: OverseerConnector,
+		args: OverseerGenArgs<'a, Spawner, RuntimeClient>,
+	) -> Result<(Overseer<Spawner, Arc<RuntimeClient>>, OverseerHandle), Error>
+	where
+		RuntimeClient: 'static + ProvideRuntimeApi<Block> + HeaderBackend<Block> + AuxStore,
+		RuntimeClient::Api: ParachainHost<Block> + BabeApi<Block> + AuthorityDiscoveryApi<Block>,
+		Spawner: 'static + SpawnNamed + Clone + Unpin,
+	{
+		let filter = StoreMaliciousAvailableData {
+			db: args.parachains_db.clone(),
+			config: args.availability_config.clone(),
+		};
+
+		prepared_overseer_builder(args)?
+			.replace_availability_store(|av| InterceptedSubsystem::new(av, filter))
+			.build_with_connector(connector)
+			.map_err(|e| e.into())
+	}
+}
+
+/// Unix time wrapper with big-endian encoding.
+#[derive(Debug, Clone, Copy, PartialEq, PartialOrd, Eq, Ord)]
+struct BETimestamp(u64);
```

Comment:
This duplicates https://github.com/paritytech/polkadot/blob/master/node/core/av-store/src/lib.rs#L78. We can reuse that to reduce the amount of code written in malus.